frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -826,9 +973,20 @@ export default function InteractiveWaveformModal({
             addPeakValley={addPeakValley}
             openChangelog={() => setOpenChangelog(true)}
             undoLastChange={undoLastChange}
-            setPeakValleyWindows={setPeakValleyWindows}
             peakValleyWindows={peakValleyWindows}
             checkDuplicates={checkDuplicates}
+            peakY1={peakY1}
+            setPeakY1={setPeakY1}
+            peakY2={peakY2}
+            setPeakY2={setPeakY2}
+            valleyY1={valleyY1}
+            setValleyY1={setValleyY1}
+            valleyY2={valleyY2}
+            setValleyY2={setValleyY2}
+            calculateYLimit={calculateYLimit}
+            twentyFourPlateDefinition={twentyFourPlateDefinition}
```

Comment:
I think this would be better imported into WaveformGraph the same way as above instead of being passed as props